### PR TITLE
Added SpanKind for gRPC Server and Client

### PR DESCRIPTION
**Pull request by siniy123**

### CODE REVIEW

File: contrib/google.golang.org/grpc/client.go [hunk 1]
```diff
@@ -29,6 +29,9 @@ func (cs *clientStream) RecvMsg(m interface{}) (err error) {
 			cs.cfg.clientServiceName(),
 			cs.cfg.analyticsRate,
 		)
+
+		span.SetTag(ext.SpanKind, ext.SpanKindClient)
```
Comment (mpetazzoni) [inline contrib/google.golang.org/grpc/client.go]: This method seems to be about the client receiving a message back from the server; the span there shouldn't be marked as kind:client. Maybe @rmfitzpatrick can help confirm.

File: contrib/google.golang.org/grpc/client.go [hunk 2]
```diff
@@ -142,6 +148,9 @@ func doClientRequest(
 		cfg.clientServiceName(),
 		cfg.analyticsRate,
 	)
+
+	span.SetTag(ext.SpanKind, ext.SpanKindClient)
```
Comment (mpetazzoni) [inline contrib/google.golang.org/grpc/client.go]: Ideally we should only get one client span per outbound interaction, not sure if we want it both here and in the SendMsg part.

File: contrib/google.golang.org/grpc/server.go [hunk 3]
```diff
@@ -34,6 +35,9 @@ func (ss *serverStream) RecvMsg(m interface{}) (err error) {
 			ss.cfg.serverServiceName(),
 			ss.cfg.analyticsRate,
 		)
+
+		span.SetTag(ext.SpanKind, ext.SpanKindServer)
```
Comment (mpetazzoni) [inline contrib/google.golang.org/grpc/server.go]: Same notes here, I think we only want one of those spans marked as kind:server, and it should be the first span capture on the server side.

**Comment by siniy123** [comment]
I removed SpanKind from Stream methods. SpanKind is only set in the interceptors.

File: contrib/google.golang.org/grpc/client.go (hunk 1 above)
Comment (rmfitzpatrick) [reply contrib/google.golang.org/grpc/client.go]: Right, I think it's for the best to only have functions that inject context into a carrier as denoting the client kind.

**Comment by jaysfx** [comment]
We haven't done a full pass to make the module supported but this change lgtm.

**Pull request merged**